src/apps/api/views/competitions.py

```diff
@@ -436,13 +437,17 @@ def get_leaderboard(self, request, pk):
         columns = [col for col in query['columns']]
         users = {}
         for submission in query['submissions']:
-            if submission['owner'] not in users.keys():
-                users.update({submission['owner']: len(users)})
+            if submission['parent']:
+                submission_key = f"{submission['owner']}{submission['parent']}"
+            else:
+                submission_key = f"{submission['owner']}{submission['id']}"
+            if submission_key not in users.keys():
+                users.update({submission_key: len(users)})
                 response['submissions'].append({'owner': submission['owner'], 'scores': []})
             for score in submission['scores']:
                 tempScore = score
                 tempScore.update({'task_id': submission['task']})
```

Comment:
Rather than adding the `task_id` to the submission score here, can we just add it in the serializer for the submission score? Then this loop could likely be turned into a simple list-comprehension.

Reply:
Right now Submission to Scores is a M2M field. So it makes sense when you aren't sure how many submissions will be attached to a score, but know that I look at this again. It's weird that this is a M2M not a One2M field. I can't think of a scenario where we would want one score attached to multiple submissions.

Reply:
Looks like a `SubmissionScore` exists for every leaderboard for every submission, so I think the M2M makes sense. Thanks for clearing that up!